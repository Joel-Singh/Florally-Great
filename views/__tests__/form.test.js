const path = require("path");
const renderPugToDOM = require(path.join(
  appRoot,
  "views",
  "renderPugToDOM.js"
));

test.each([
  ["Form renders", {}],
  [
    "Form renders a single error",
    { errors: [{ msg: "This Error Msg Should be seen" }] },
  ],
  [
    "Form renders multiple errors",
    {
      errors: [
        { msg: "This Error Msg Should be seen 1" },
        { msg: "This Error Msg Should be seen 2" },
        { msg: "This Error Msg Should be seen 3" },
      ],
    },
  ],
])("%s", (name, locals) => {
  const form = renderPugToDOM("./views/form.pug", locals).querySelector("form");
  expect(form).toMatchSnapshot();
});
