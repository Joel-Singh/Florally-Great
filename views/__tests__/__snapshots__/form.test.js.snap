// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
<form
  action=""
  method="POST"
/>
`;

exports[`Form renders a single error 1`] = `
<form
  action=""
  method="POST"
>
  <ul
    data-testid="errors"
  >
    <li>
      This Error Msg Should be seen
    </li>
  </ul>
</form>
`;

exports[`Form renders multiple errors 1`] = `
<form
  action=""
  method="POST"
>
  <ul
    data-testid="errors"
  >
    <li>
      This Error Msg Should be seen 1
    </li>
    <li>
      This Error Msg Should be seen 2
    </li>
    <li>
      This Error Msg Should be seen 3
    </li>
  </ul>
</form>
`;
